src/components/ConditionalFilter/API/initialState/product/useProductInitialAPIState.tsx

```diff
@@ -87,21 +104,100 @@ export const useProductInitialAPIState = (): InitialProductAPIState => {
       );
     }
 
-    if (Object.keys(attribute).length > 0) {
+    // Fetch attribute definitions for both regular and reference attributes
+    // (We need the definitions for URL parsing, but values are fetched separately)
+    const allAttributeSlugs = [...Object.keys(attribute), ...Object.keys(attributeReference)];
+
+    if (allAttributeSlugs.length > 0) {
+      // Only fetch choices values for regular attributes, not reference attributes
+      // (if we try fetching reference attribute options Saleor returns 500 error)
+      const regularChoiceIds = Object.values(attribute).flat().filter(Boolean);
+
       queriesToRun.push(
         client.query<_SearchAttributeOperandsQuery, _SearchAttributeOperandsQueryVariables>({
           query: _SearchAttributeOperandsDocument,
           variables: {
-            attributesSlugs: Object.keys(attribute),
-            choicesIds: Object.values<string[]>(attribute).flat(),
-            first: Object.keys(attribute).length,
+            attributesSlugs: allAttributeSlugs,
+            choicesIds: regularChoiceIds,
+            first: allAttributeSlugs.length,
           },
         }),
       );
     }
 
     const data = await Promise.all(queriesToRun);
-    const initialState = createInitialProductStateFromData(data, channel);
+    let initialState = createInitialProductStateFromData(data, channel);
+
+    // Fetch Choices for Reference Attributes
```

Comment:
fyi: this needs to be done **after** we fetch data about attributes, because we need to know `entityType` from attribute's definition